src/TableFormatter.ts

```diff
@@ -1,31 +1,37 @@
-import { Editor } from "codemirror";
-import formatTable from "../lib/FormatTable";
+import {Editor} from 'codemirror'
+import formatTable from '../lib/FormatTable'
+import prettier from 'prettier'
 
 module.exports = {
-	default: function(_context: any) {
+  default: function (_context: any) {
 
-		const plugin = function(CodeMirror) {
-			CodeMirror.defineExtension('formatTable', function() {
-				const cm: Editor = this; // to get autocomplete working
+    const plugin = function (CodeMirror) {
+      CodeMirror.defineExtension('formatTable', function () {
+        const editor: Editor = this // to get autocomplete working
 
-				const cursor = cm.getCursor();
-				let startLine = cursor.line;
-				while (startLine >=0 && cm.getLine(startLine).trimStart().charAt(0) === '|') startLine--;
-				startLine++;
+        let parser = 'markdown' // for JS. Other values: json, typescript, css, scss, less, html (also for xml), yaml or php
 
-				let endLine = cursor.line;
-				while (!!cm.getLine(endLine) && cm.getLine(endLine).trimStart().charAt(0) === '|') endLine++;
-				
-				let formatted = formatTable(startLine, endLine - startLine, i => cm.getLine(i));
```

Comment:
if this function is no longer needed, why keep the file and tests?

Reply:
This is just an example, so I haven't dealt with many details (because I don't use joplin editor, vscode can be set to use prettier to format markdown)